scripts/code_review_tool_evaluator.py

```diff
@@ -29,6 +31,189 @@
 from bugbug.vectordb import QdrantVectorDB
 
 code_review.TARGET_SOFTWARE = "Mozilla Firefox"
+VERBOSE_CODE_REVIEW = False
+
+
+EVALUATION_TEMPLATE = """Your are an expert in code review at Mozilla Firefox.
+
+**Task**:
+
+Match two sets of code review comments to identify redundant comments.
+
+**Instructions**:
+
+    1. **Consider the following about all comments**:
+        - The comments are related to the same code patch.
+        - The comments may be written in different styles.
+
+    2. **Understand what each comment is addressing**:
+        - Read the comments in both sets.
+        - Understand the issue that each comment is addressing.
+
+    3. **Check for matches**:
+        - If you find a comment in the old set that is addressing the same issue as a comment in the new set, link them as redundant.
+        - The comments may not be identical, but they should be addressing the same issue.
+        - The level of detail in the comments may vary.
+
+    4. **Output format**:
+        - Output a list of matched comments.
+        - Include the comment IDs only in the output.
+        - Each element in the list should be an object with two keys: `old_comment_id` and `new_comment_id`.
+        - No explanation is needed in the output, only the IDs of the matched comments.
+        - The output should be a valid json only.
+
+
+**Output example**:
+
+    [
+        {{"old_comment_id": 1, "new_comment_id": 3}},
+        {{"old_comment_id": 4, "new_comment_id": 2}},
+    ]
+
+**First set of comments (old comments)**:
+
+{old_comments}
+
+**Second set of comments (new comments)**:
+
+{new_comments}
+"""
+
+
+class FeedbackEvaluator:
+    def __init__(self, evaluation_dataset: str):
+        self.evaluated_comments = pd.read_csv(evaluation_dataset)
+
+        llm = generative_model_tool.create_openai_llm()
+        evaluate_comments_prompt = PromptTemplate.from_template(EVALUATION_TEMPLATE)
+        self.evaluation_chain = evaluate_comments_prompt | llm
```

Comment:
I really don't like this very specific langchain stuff :(
We can change it later though

Reply:
Fixed here: https://github.com/mozilla/bugbug/compare/ab04848e8bc7dba4c0e42cf0e99f93d28c198876..2bc317d3b4d992b49bdd8a92835d258455b4c4c7